src/app/core/parameters/segmentArray.ts

```diff
@@ -0,0 +1,30 @@
+import Parameter from "../parameter";
+import Registry from "../registry";
+
+export default class SegmentArray extends Parameter {
+    typeString: string = "SegmentArray";
+    description: string = "Should be an array of PointArrays";
+
+    constructor(value: Array<Array<Array<number>>>) {
+        super("SegmentArray",value);
+    }
+
+    isValid(value: any) {
+        if (value instanceof Array) {
+            if (value.length == 0) {
+                return true;
+            }
+            /*for (var i in value) {
+                let paramType = Registry.registeredParams["PointArray"];
```

Comment:
Import point array from the typescript file instead of Registry here like in the previous one.